csp/constants.py

```diff
@@ -15,12 +17,12 @@
 class Nonce:
     _instance = None
 
-    def __new__(cls, *args, **kwargs):
+    def __new__(cls: Type["Nonce"], *args: Any, **kwargs: Any) -> "Nonce":
```

Comment:
the way to write this is:

```python
def __new__(cls, *args: Any, **kwargs: Any) -> Self:
```

(`Self` is importable from `typing_extensions` for the pythons supported here)

Reply:
Yes, this seems to work. Do you want to submit the PR?